api/src/main/java/io/minio/errors/ErrorResponseException.java

```diff
@@ -57,7 +57,9 @@ public String toString() {
         + "request={"
         + "method=" + request.method() + ", "
         + "url=" + request.httpUrl() + ", "
-        + "headers=" + request.headers()
+        + "headers=" + request.headers().toString()
+                .replaceAll("Signature=([0-9a-f]+)", "Signature=*REDACTED*")
+                .replaceAll("Credential=([^/]+)", "Credential=*REDACTED*")
```

Comment:
Credential is part of the Authorization header @balamurugana . This while making the request not what s3 sent back.